SYCL/InlineAsm/asm_no_operands.cpp

```diff
@@ -12,8 +12,14 @@ int main() {
   cl::sycl::queue Queue;
   cl::sycl::device Device = Queue.get_device();
 
+  std::string extensions;
+  if (Device.get_backend() == cl::sycl::backend::opencl)
+    extensions = get_extensions_opencl(Device);
+
   if (!isInlineASMSupported(Device) ||
-      !Device.has_extension("cl_intel_required_subgroup_size")) {
+      (Device.get_backend() != cl::sycl::backend::level_zero &&
+       extensions.find("cl_intel_required_subgroup_size") ==
+           std::string::npos)) {
```

Comment:
I would expect that we have a device aspect, which corresponds to this extension and if that is the case, then it would be better to check it instead of going directly to OpenCL.

Moreover, `reqd_sub_group_size` is a core SYCL 2020 functionality, which should be supported everywhere, so, probably we can omit the check completely then?

The same logic probably applies to the rest of tests modified in this PR, but I would like to hear from @Pennycook to see what John says

Reply:
I agree, there's no need to check for `cl_intel_required_subgroup_size` or any other extensions related to sub-group capabilities.  Sub-group support is not optional in SYCL 2020.

I suspect we might see some new failures as a result of this change, because there might be some devices that don't have complete sub-group support yet.  But it makes more sense to me to try and address those cases by updating the UNSUPPORTED/XFAIL tags than to add logic that skips the tests.

Reply:
Do you mind if we investigate it separately to avoid blocking deprecation activity?

Reply:
> Do you mind if we investigate it separately to avoid blocking deprecation activity?

I actually don't expect that we have devices in our CI that don't support sub-groups, so it worth to try, IMHO. If you decide to postpone, then could you please create a tracker for that?

Reply:
> I actually don't expect that we have devices in our CI that don't support sub-groups,

There might be more devices that we would like to run these tests on than available in CI.
Maxim, could you please create a tracker?

Reply:
New approach with `get_info<device::extensions>` looks good to me, so I think it is not on a critical path to try and remove those checks.

Reply:
They will be removed with removing device::info::extensions